diff --git dwm.c dwm.c
index e63e623..6511464 100644
--- dwm.c
+++ dwm.c
@@ -1746,10 +1746,20 @@ void
 toggleview(const Arg *arg)
 {
 	unsigned int newtagset = selmon->tagset[selmon->seltags] ^ (arg->ui & TAGMASK);
-
-	if (newtagset) {
-		selmon->tagset[selmon->seltags] = newtagset;
-		focus(NULL);
+  Client *c = NULL;
+  unsigned int found = 0;
+
+  if (newtagset) {
+    selmon->tagset[selmon->seltags] = newtagset;
+
+    for(c = selmon->clients; c && !(found = c->tags & arg->ui); c = c->next);
+    if(found && ISVISIBLE(c)) {
+      detach(c);
+      attach(c);
+      focus(c);
+    }
+    else
+      focus(NULL);
 		arrange(selmon);
 	}
 }
